.github/CONTRIBUTING.md

```diff
@@ -1,4 +1,47 @@
-# How to Contribute
+# Contributing
+
+This library's source code is mainly generated from JSON schema in the [Google CloudEvents repo](https://github.com/googleapis/google-cloudevents).
+
+Specifically, the `google-cloudevents-types/src` directory is generated and should not be modified as changes will be overwritten when the library is re-generated. Rather, the upstream CloudEvents repo should be modified.
+
+## Generate API types
+
+Before generating the library, you must set up your environment like such:
+- Have Java 11+ and Node (for generation) installed.
+- Install the [quicktype wrapper CLI](https://github.com/googleapis/google-cloudevents/tree/master/tools/quicktype-wrapper) using npm.
+- Clone the [google-cloudevents](https://github.com/googleapis/google-cloudevents) source of truth repo in the same directory as this repo.
```

Comment:
This seems odd. Are you adding then removing it from this repo?

Reply:
- We're cloning the JSON schema files locally so we can generate java files using the schemas.
- The repo is in the same directory as this repo, `google-cloudevents-java`
- We're not removing the schemas.

Reply:
It feels like using a submodule might make this simpler - and then in particular, the generated code could be kept in sync with the current submodule commit, but I'm not overly bothered.
(I haven't checked, but I'd hope that google-cloudevents/ is in .gitignore. Users shouldn't need to remove it afterwards)